lib/ceedling/configurator.rb

```diff
@@ -111,13 +113,10 @@ def populate_cmock_defaults(config)
 
     if (cmock[:unity_helper])
       cmock[:unity_helper] = [cmock[:unity_helper]] if cmock[:unity_helper].is_a? String
+      cmock[:includes] = [] if (cmock[:includes].nil?)
```

Comment:
I think this is needed because the empty `includes` array is in the `default_config` now. I'm concerned there might be other places (including perhaps in plugins and user code?) where not having the empty arrays in `config_hash` until after the defaults are merged might cause problems.

Reply:
Checked all other places where config is added to before the defaults are merged, none of them access empty arrays except config[:plugins][:enabled] and config[:plugins][:load_paths] which are both initialised in CEEDLING_CONFIG_INTERNAL. Plugins and user code are not run until after the defaults are merged.